client/src/app/special-accounts/edit/special-account-edit.component.ts

```diff
@@ -0,0 +1,209 @@
+        });
+    }
+
+    prepareEditing(): SpecialAccount {
+        const specialAccount = new SpecialAccount({
+            id: this.currentSpecialAccount.id,
+        });
+
+        const formValues = this.specialAccountForm.value;
+
+        // Simple field
+        if (this.currentSpecialAccount.description !== formValues.description) {
+            specialAccount.description = formValues.description;
+        }
+        if (this.currentSpecialAccount.connection.username !== formValues.username) {
+            specialAccount.connection = {
+                username: formValues.username
+            };
+        }
+        if (formValues.password) {
+            specialAccount.connection = {
+                ...specialAccount.connection,
+                password: formValues.password
+            };
+        }
+        if (formValues.code) {
+            specialAccount.code = formValues.code;
+        }
+
+        // Associations
+        const add = this.permissions.filter(permission => {
+            return permission.isChecked === true && permission.initial !== permission.isChecked;
+        });
+        const remove = this.permissions.filter(permission => {
+            return permission.isChecked === false && permission.initial !== permission.isChecked;
+        });
+        if (add.length > 0) {
+            specialAccount._embedded = {
+                permissions: {
+                    add: add.map(perm => perm.permission.id),
+                }
+            };
+        }
+        if (remove.length > 0) {
+            if (specialAccount._embedded) {
+                specialAccount._embedded.permissions.remove = remove.map(perm => perm.permission.id);
+            } else {
+                specialAccount._embedded = {
+                    permissions: {
+                        remove: remove.map(perm => perm.permission.id),
+                    }
+                };
+            }
+        }
+        return specialAccount;
+    }
+
+    disable(): Boolean {
+        const add = this.permissions.filter(permission => {
+            return permission.isChecked === true && permission.initial !== permission.isChecked;
+        });
+        const remove = this.permissions.filter(permission => {
+            return permission.isChecked === false && permission.initial !== permission.isChecked;
+        });
+        if (!this.specialAccountForm.valid) {
+            return true;
+        }
+        return this.currentSpecialAccount.connection.username === this.specialAccountForm.get('username').value
+            && this.currentSpecialAccount.description === this.specialAccountForm.get('description').value
+            && add.length === 0
+            && remove.length === 0
+            && this.specialAccountForm.get('code').value === '';
+    }
+
+    isMe(): Boolean {
+        if (this.currentUser && this.currentUser.specialAccount) {
+            return this.currentSpecialAccount.id === this.currentUser.specialAccount.id ? true : false;
```

Comment:
We could simplify this to `return this.currentUser && this.currentUser.specialAccount && 
 this.currentUser.specialAccount.id === this.currentSpecialAccount.id` ?

Reply:
Thanks WebStorm :+1: